pkg/cmd/registry/rule/enable/enable.go

```diff
@@ -0,0 +1,268 @@
+		return err
+	}
+
+	dataAPI, _, err := conn.API().ServiceRegistryInstance(opts.registryID)
+	if err != nil {
+		return err
+	}
+
+	rule := registryinstanceclient.Rule{
+		Config: rulecmdutil.GetMappedConfigValue(opts.config),
+		Type:   rulecmdutil.GetMappedRuleType(opts.ruleType),
+	}
+
+	if opts.artifactID == "" {
+
+		opts.Logger.Info(opts.localizer.MustLocalize(
+			"registry.rule.enable.log.info.enabling.globalRules",
+			localize.NewEntry("RuleType", opts.ruleType),
+			localize.NewEntry("Configuration", opts.config)))
+
+		req := dataAPI.AdminApi.CreateGlobalRule(opts.Context)
+
+		req = req.Rule(rule)
+
+		httpRes, newErr = req.Execute()
+		if httpRes != nil {
+			defer httpRes.Body.Close()
+		}
+	} else {
+
+		opts.Logger.Info(opts.localizer.MustLocalize(
+			"registry.rule.enable.log.info.enabling.artifactRules",
+			localize.NewEntry("RuleType", opts.ruleType),
+			localize.NewEntry("Configuration", opts.config),
+			localize.NewEntry("ArtifactID", opts.artifactID)))
+
+		req := dataAPI.ArtifactRulesApi.CreateArtifactRule(opts.Context, opts.group, opts.artifactID)
+
+		req = req.Rule(rule)
+
+		httpRes, newErr = req.Execute()
+		if httpRes != nil {
+			defer httpRes.Body.Close()
+		}
+	}
+
+	ruleErr := &rulecmdutil.RegistryRuleError{
+		Localizer:  opts.localizer,
+		InstanceID: opts.registryID,
+	}
+
+	if newErr != nil {
+		if httpRes == nil {
+			return registrycmdutil.TransformInstanceError(newErr)
+		}
+
+		switch httpRes.StatusCode {
+		case http.StatusNotFound:
+			return ruleErr.ArtifactNotFoundError(opts.artifactID)
+		case http.StatusConflict:
+			return ruleErr.ConflictError(opts.ruleType)
+		default:
+			return registrycmdutil.TransformInstanceError(newErr)
+		}
+
+	}
+
+	opts.Logger.Info(icon.SuccessPrefix(), opts.localizer.MustLocalize("registry.rule.enable.log.info.ruleEnabled"))
+
+	return nil
+}
+
+func runInteractivePrompt(opts *options) (err error) {
+
+	ruleTypePrompt := &survey.Select{
+		Message: opts.localizer.MustLocalize("registry.rule.enable.input.ruleType.message"),
+		Help:    opts.localizer.MustLocalize("registry.rule.common.flag.ruleType"),
+		Options: rulecmdutil.ValidRuleTypes,
+	}
+
+	err = survey.AskOne(ruleTypePrompt, &opts.ruleType)
+	if err != nil {
+		return err
+	}
+
+	configOptions := rulecmdutil.ValidRuleConfigs[opts.ruleType]
+
+	configPrompt := &survey.Select{
+		Message: opts.localizer.MustLocalize("registry.rule.enable.input.config.message"),
+		Help:    opts.localizer.MustLocalize("registry.rule.common.flag.config"),
+		Options: configOptions,
+	}
+
+	err = survey.AskOne(configPrompt, &opts.config)
+	if err != nil {
+		return err
+	}
+
+	instanceIDPrompt := &survey.Input{
+		Message: opts.localizer.MustLocalize("registry.rule.enable.input.instanceID.message"),
+		Help:    opts.localizer.MustLocalize("registry.rule.enable.input.instanceID.help"),
+	}
+
+	err = survey.AskOne(instanceIDPrompt, &opts.registryID)
+	if err != nil {
+		return err
+	}
+
+	artifactIDPrompt := &survey.Input{
```

Comment:
Validating and suggesting values becomes bit tricky here as artifact operations need the group before hand. Even if we put the group prompt before artifact prompt, there doesn't seem to be any way to retrieve the list of groups.

Reply:
> I don't think we should be prompting for the instance ID - it is not a pattern used in the other commands.

I agree but other interactive modes didn't have the instance id flags, I am not sure if we should make the interactive mode have less options than non-interactive.

Reply:
Let's keep in mind that sometimes reasons why integration is soo hard is because API is simply making it hard. If there is some suggestion we can put for future versions of the API (this area is being worked on) it would be good for entire service.

Reply:
Changed instance ID argument to flag only, removed from prompt.